src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java

```diff
@@ -166,12 +172,15 @@ private Map<String, Object> getResponseMap(String baseUrl) throws IOException {
           StandardCharsets.UTF_8);
         return (Map<String, Object>) GSON.fromJson(responseAsString, Map.class);
       }
+      LOG.error("Unable to fetch the response.");
       throw new HttpResponseException(statusCode, "No response.");
     }
   }
 
-  List<String> getJsonValuesFromResponse(Map<String, Object> responseMap, ObjectType object) {
+  private List<String> getJsonValuesFromResponse(Map<String, Object> responseMap, ObjectType object) {
     List<Object> responseObjects = (List<Object>) responseMap.get(object.getResponseKey());
+    //checking for the childKey, some objects are dependent on the other objects, so data of those objects is being
+    // fetched on the basis of the keys of the object on which it is dependent on.
```

Comment:
if childKey is present, why are we filtering only the event_type in comments? Please update the comment such that someone who is not familiar with Zendesk can understand why you are doing this

Reply:
please update

Reply:
done